pallets/time-release/src/tests.rs

```diff
@@ -415,6 +415,230 @@ fn exceeding_maximum_schedules_should_fail() {
+		assert_eq!(
+			PalletBalances::balance_frozen(&FreezeReason::TimeReleaseVesting.into(), &BOB),
+			20u64
+		);
+	});
+}
+
+#[test]
+fn multiple_schedule_transfer_success() {
+	ExtBuilder::build().execute_with(|| {
+		let schedule = ReleaseSchedule {
+			start: 6u32.into(),
+			period: 10u32.into(),
+			period_count: 2u32,
+			per_period: 10u32.into(),
+		};
+
+		let when = BlockNumberFor::<Test>::from(4u32);
+
+		assert_ok!(TimeRelease::schedule_transfer(
+			RuntimeOrigin::signed(ALICE),
+			BOB,
+			schedule,
+			when
+		));
+		let schedules = pallet_scheduler::Agenda::<Test>::get(4u32);
+		assert_eq!(schedules.len(), 1);
+		let hold_balance = PalletBalances::balance_on_hold(
+			&HoldReason::TimeReleaseScheduledVesting.into(),
+			&ALICE,
+		);
+		assert_eq!(hold_balance, 20u64);
+
+		// Adding another schedule increases the hold
+
+		let schedule_2 = ReleaseSchedule {
+			start: 6u32.into(),
+			period: 10u32.into(),
+			period_count: 2u32,
+			per_period: 10u32.into(),
+		};
+
+		let when = BlockNumberFor::<Test>::from(4u32);
+
+		assert_ok!(TimeRelease::schedule_transfer(
+			RuntimeOrigin::signed(ALICE),
+			BOB,
+			schedule_2,
+			when
+		));
+		let schedules = pallet_scheduler::Agenda::<Test>::get(4u32);
+		assert_eq!(schedules.len(), 2);
+		let hold_balance = PalletBalances::balance_on_hold(
+			&HoldReason::TimeReleaseScheduledVesting.into(),
+			&ALICE,
+		);
+		assert_eq!(hold_balance, 40u64);
+	});
+}
+
+#[test]
+fn schedule_transfer_fails_if_invalid_block_number() {
+	ExtBuilder::build().execute_with(|| {
+		let schedule = ReleaseSchedule {
+			start: 6u32.into(),
+			period: 10u32.into(),
+			period_count: 2u32,
+			per_period: 10u32.into(),
+		};
+		let invalid_block_number = 0u32;
+		assert_noop!(
+			TimeRelease::schedule_transfer(
+				RuntimeOrigin::signed(ALICE),
+				BOB,
+				schedule,
+				invalid_block_number
+			),
+			pallet_scheduler::Error::<Test>::TargetBlockNumberInPast
+		);
+	});
+}
+
+#[test]
+fn schedule_transfer_fails_if_bad_origin() {
+	ExtBuilder::build().execute_with(|| {
+		let schedule = ReleaseSchedule {
+			start: 6u32.into(),
+			period: 10u32.into(),
+			period_count: 2u32,
+			per_period: 10u32.into(),
+		};
+		let when = BlockNumberFor::<Test>::from(4u32);
+		assert_noop!(
+			TimeRelease::schedule_transfer(RuntimeOrigin::signed(CHARLIE), BOB, schedule, when),
+			BadOrigin
+		);
+	});
+}
+
+#[test]
+fn execute_scheduled_transfer_success() {
+	ExtBuilder::build().execute_with(|| {
+		let schedule = ReleaseSchedule {
+			start: 6u32.into(),
+			period: 10u32.into(),
+			period_count: 2u32,
+			per_period: 10u32.into(),
+		};
+		let when = BlockNumberFor::<Test>::from(4u32);
+		assert_ok!(TimeRelease::schedule_transfer(
+			RuntimeOrigin::signed(ALICE),
+			BOB,
+			schedule.clone(),
+			when
+		));
+
+		assert_ok!(TimeRelease::execute_scheduled_transfer(
+			Origin::<Test>::TimeRelease(ALICE).into(),
+			BOB,
+			schedule
+		));
+		assert_eq!(PalletBalances::free_balance(BOB), 20u64);
```

Comment:
These tests could use comments reminding a reader how much hold/freeze balance these accounts have.  

Also it seems like if there are 2 periods I would expect 2 scheduled transfers. But I haven't looked at this pallet in a really long time.

Reply:
There is only 1 transfer here.